fMRIVolume/GenericfMRIVolumeProcessingPipeline.sh

```diff
@@ -626,145 +649,168 @@ fi
 
 ########################################## DO WORK ##########################################
 
-T1wFolder="$Path"/"$Subject"/"$T1wFolder"
-AtlasSpaceFolder="$Path"/"$Subject"/"$AtlasSpaceFolder"
-ResultsFolder="$AtlasSpaceFolder"/"$ResultsFolder"/"$NameOffMRI"
+T1wFolderCross="$Path"/"$Session"/"$T1wFolder"
+T1wFolderLong="$Path"/"$SessionLong"/"$T1wFolder"
+T1wFolder=$T1wFolderCross
 
-mkdir -p ${T1wFolder}/Results/${NameOffMRI}
 
-if [ ! -e "$fMRIFolder" ] ; then
-    log_Msg "mkdir ${fMRIFolder}"
-    mkdir "$fMRIFolder"
-fi
-${FSLDIR}/bin/imcp "$fMRITimeSeries" "$fMRIFolder"/"$OrigTCSName"
+AtlasSpaceFolderCross="$Path"/"$Session"/"$AtlasSpaceFolder"
+AtlasSpaceFolderLong="$Path"/"$SessionLong"/"$AtlasSpaceFolder"
+AtlasSpaceFolder=$AtlasSpaceFolderCross
 
+ResultsFolderCross="$AtlasSpaceFolder"/"$ResultsFolder"/"$NameOffMRI"
+ResultsFolderLong="$AtlasSpaceFolderLong"/"$ResultsFolder"/"$NameOffMRI"
+ResultsFolder=$ResultsFolderCross
 
-if [[ $nEcho -gt 1 ]] ; then
-    log_Msg "$nEcho TE's supplied, running in multi-echo mode"
-    NumFrames=$("${FSLDIR}"/bin/fslval "${fMRIFolder}/${OrigTCSName}" dim4)
-    FramesPerEcho=$((NumFrames / nEcho))
-    EchoDir="${fMRIFolder}/MultiEcho"
-    mkdir -p "$EchoDir"
+if (( ! IsLongitudinal )); then 
+    mkdir -p ${T1wFolder}/Results/${NameOffMRI}
+    if [ ! -e "$fMRIFolder" ] ; then
+        log_Msg "mkdir ${fMRIFolder}"
+        mkdir "$fMRIFolder"
+    fi
+else
+    #copy directory structure and create symbolic link per each file under source. 
+    cp -rf "$ResultsFolder" "$ResultsFolderLong"
+    if (( $? )); then
+        log_Err_Abort "Copying cross-sectional output $ResultsFolder to longitudinal session folder $ResultsFolderLong failed."
+    fi
+    fMRIFolderLong="$Path"/"$SessionLong"/"$NameOffMRI"
+    cp -rf "$fMRIFolder" "$fMRIFolderLong"
+	if (( $? )); then
+		log_Err_Abort "Copying cross-sectional output $fMRIFolder to longitudinal session folder $fMRIFolderLong failed."
+	fi
```

Comment:
```suggestion
    cp -rf "$ResultsFolder" "$ResultsFolderLong"
    fMRIFolderLong="$Path"/"$SessionLong"/"$NameOffMRI"
    cp -rf "$fMRIFolder" "$fMRIFolderLong"
```
Under `set -eu` and `debug.shlib`, all nonzero exit statuses will be caught before they reach your conditionals.  If you really feel these custom messages are necessary, you would need to do it this way:

```
    if ! cp -rf "$ResultsFolder" "$ResultsFolderLong"
    then
        log_Err_Abort "Copying cross-sectional output $ResultsFolder to longitudinal session folder $ResultsFolderLong failed."
    fi
```

You need whitespace on both sides of the `!`, due to bash history syntax also using that symbol.

The second conditional also somehow ended up with tabs instead of spaces.

Reply:
Ok